pkg/api/validation/validation.go

```diff
@@ -634,6 +657,8 @@ func validateEnv(vars []api.EnvVar) errs.ValidationErrorList {
 	return allErrs
 }
 
+var validFieldPathExpressions = util.NewStringSet("metadata.name", "metadata.namespace")
```

Comment:
`validateMetadataVolumeSource` is validate against  `validMetadataFieldPathExpressions` not against `validFieldPathExpressions`. I had to duplicate the two lists cause https://github.com/GoogleCloudPlatform/kubernetes/tree/master/examples/downward-api support only `name` and `namespace`